app/code/core/Mage/Core/Model/Config.php

```diff
@@ -501,13 +491,43 @@ public function getCache()
     }
 
     /**
-     * Get lock flag cache identifier
+     * Call before building and saving cache to ensure only one process can save the cache
      *
-     * @return string
+     * If failed to get cache lock:
+     *   - CLI: throws exception
+     *   - Other: 503 error
+     *
+     * @return void
+     * @throws Exception
      */
-    protected function _getCacheLockId()
+    public function getCacheSaveLock()
     {
-        return $this->getCacheId() . '.lock';
+        if (!Mage::app()->useCache('config')) {
+            return;
+        }
+        $connection = Mage::getSingleton('core/resource')->getConnection('core_write');
+        if (!$connection->fetchOne("SELECT GET_LOCK('core_config_cache_save_lock', ?)", [PHP_SAPI === 'cli' ? 60 : 3])) {
+            if (PHP_SAPI === 'cli') {
+                throw new Exception('Could not get lock on cache save operation.');
+            } else {
+                require_once Mage::getBaseDir() . DS . 'errors' . DS . '503.php';
```

Comment:
```suggestion
                require Mage::getBaseDir() . DS . 'errors' . DS . '503.php';
```

it will never happen more than once but I think this shouldn't be a require_once

Reply:
Is there a PSR or something that recommends using require instead of require_once? I don't care either way, just curious.:)

Reply:
it's just because if that file was already "required" before, then the content wouldn't run and you'd just jump to the `die()` so I think it's more correct to have a proper require there.